File: api/src/main/java/api/security/model/ErrorResponse.java
package api.security.model;

import org.springframework.http.HttpStatus;

import java.util.Date;

public class ErrorResponse {

    private final HttpStatus status;
    private final String message;
    private final Date timestamp;

    public ErrorResponse(HttpStatus status, String message, Date timestamp) {
        this.status = status;
        this.message = message;
        this.timestamp = timestamp;
    }

    public HttpStatus getStatus() {
        return status;
    }

    public String getMessage() {
        return message;
    }

    public Date getTimestamp() {
        return timestamp;
    }
}

File: api/src/main/java/api/security/model/ErrorResponse.kt
package api.security.model

import org.springframework.http.HttpStatus
import java.time.LocalDateTime

class ErrorResponse(val status: HttpStatus, val message: String?){

    val errorCode = status.value()
    val timestamp = LocalDateTime.now()

    constructor(exception: Exception?) :
            this(HttpStatus.INTERNAL_SERVER_ERROR, exception?.message)

}
